### ISSUE TITLE
spark writer: task failure because s3 eventual consistency during HadoopInputFile.getStat

### ISSUE THREAD

**Issue by sudssf** [apache/incubator-iceberg#881]
We see few tasks failures for large tables because of s3 eventual consistency as getStatus us called immediately after creating file.
after checking s3 , file does exists. we have observed this issue in non-spark client as well.
one possible solution here is to use writer.length to get file size instead of calling getStatus

workaround (https://github.com/apache/incubator-iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/source/Writer.java#L443)
```
//replace
.withEncryptedOutputFile(currentFile)
//with
.withEncryptionKeyMetadata(currentFile.keyMetadata())
.withPath(currentFile.encryptingOutputFile().location())
.withFileSizeInBytes(currentAppender.length())    
```
stacktrace
```
ERROR Utils: Aborting task
org.apache.iceberg.exceptions.RuntimeIOException: Failed to get status for file: s3a://...
	at org.apache.iceberg.hadoop.HadoopInputFile.lazyStat(HadoopInputFile.java:110)
	at org.apache.iceberg.hadoop.HadoopInputFile.getStat(HadoopInputFile.java:140)
	at org.apache.iceberg.DataFiles$Builder.withInputFile(DataFiles.java:245)
	at org.apache.iceberg.DataFiles$Builder.withEncryptedOutputFile(DataFiles.java:254)
	at org.apache.iceberg.spark.source.Writer$BaseWriter.closeCurrent(Writer.java:445)
	at org.apache.iceberg.spark.source.Writer$BaseWriter.commit(Writer.java:405)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$$anonfun$run$3.apply(WriteToDataSourceV2Exec.scala:127)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$$anonfun$run$3.apply(WriteToDataSourceV2Exec.scala:116)
	at org.apache.spark.util.Utils$.tryWithSafeFinallyAndFailureCallbacks(Utils.scala:1394)
	at org.apache.spark.sql.execution.datasources.v2.DataWritingSparkTask$.run(WriteToDataSourceV2Exec.scala:146)
	at org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2Exec$$anonfun$doExecute$2.apply(WriteToDataSourceV2Exec.scala:67)
	at org.apache.spark.sql.execution.datasources.v2.WriteToDataSourceV2Exec$$anonfun$doExecute$2.apply(WriteToDataSourceV2Exec.scala:66)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.FileNotFoundException: No such file or directory: s3a://...
	at org.apache.hadoop.fs.s3a.S3AFileSystem.getFileStatus(S3AFileSystem.java:1642)
	at org.apache.hadoop.fs.s3a.S3AFileSystem.getFileStatus(S3AFileSystem.java:117)
	at org.apache.iceberg.hadoop.HadoopInputFile.lazyStat(HadoopInputFile.java:108)
	... 19 more
```

**Comment by sudssf** [comment]
submitted https://github.com/apache/incubator-iceberg/pull/882

**Comment by rdblue** [comment]
Closing this, fixed by #882.
